---
layout: default
order: 2
title:  "Week 2: Shell scripting"
date: "2025-01-21"
time: ""
categories: main
instructor: 
lesson: https://github.com/pesalerno/computing_SP25/edit/master/2025-01-21-2_Week_2.md
material-type: 
lesson-type: yes
---

<a href="https://pesalerno.github.io/computing_SP25/main/2025/01/14/1_Week_1.html"><button>Previous Lesson</button></a>    <a href="https://pesalerno.github.io/computing_SP25/main/2025/01/28/3_Week_3.html"><button>Next Lesson</button></a>

# Week 2 | Shell scripting and Illumina data

------------
> Week's goals: 
>
>- More shell/understanding your computer
>
>- Working with Illumina data & viewing/editing large text files
>

--------------


> **SUMMARY:** This week, we will be working with large text files in the shell, as well as with getting more practice with command line and moving around directories/files using the shell. First, let's review Illumina sequencing and learn a bit about multiplexing in large genomics projects. 





##There are a few important concepts to know about Illumina sequencing: 

- Chips contain flowcells where the DNA is "hybridized" in a process known as [***"Sequence by Synthesis"***](https://www.youtube.com/watch?v=fCd6B5HRaZ8)

- Illumina sequencing (and all other next-gen sequencing platforms) is, by nature, highly error prone. Thus, the first step in any next-gen sequencing project is to make sure we only keep sequences that are of sufficient quality. For this, we use the program `fastqc`

- Because a single flow-cell is usually too much data for a single individual in most genomics projects, Illumina projects are often **multiplexed** - where many individual samples are barcoded and pooled into a single sample. This is later resolved computationally, by extracting (demultiplexing) sequences that contain the barcode in a specific place. 

- Although we could write a somewhat straightforward program for demultiplexing using the command line, many programs have already been written for this, taking into account many forms of error that can occur. Further, due to the numerous forms of error in Illumina, and to the even higher potential for error in reconstructing highly repetitive eukaryotic genomes, the pipelines for analyzing, filtering, and outputting genomics datasets can get extremely complex, for example, in protocols such as [denovo ddRAD genome assemblies](https://github.com/pesalerno/computing_SP25/tree/main/_files/denovo-pipeline.png), which are commonly used to assess conservation genomics status of threatened species in the wild. 


## Downloading our data 

First, let's start with downloading the data using the following commands: 

```bash	

	cd go/to/your/class/directory
	curl -L -o Bjimenezi_Churi2.fq.gz "https://drive.usercontent.google.com/download?id={1MLg0Mx5yrWqnwBKzUslk3IL-1GrKKp7I}&confirm=xxx"
```

You need to run the above code several times to download all the data files we will be using this and next week, replacing the `{FILEID}` and corresponding output file names with the following ones: 

| `{FILEID}` | `-o` `*.fq.gz` |
| --- | --- |
| 1ZKk9xGdzsWD6Nhi_vYU-eTowx6FraGtM | Bjimenezi_Churi2 |
| 1IBPiKpXOwlsa58M5uRfOsWog-S-2sdoF | Bjimenezi_Guaya1 |
| 1nCs_9qmUJE4sufFxEC57Fwe4yw62LqGg | Bjimenezi_Guaya2 |
| 14kG5096rQ-tuOj_aaour6UW_Dbwu83nL | Bjimenezi_Oso1 |
| 1YfKcJYsiZyyhyTUybLZx25QFCtmfPpFE | Bjimenezi_Oso2 |









<a href="https://pesalerno.github.io/computing_SP25/main/2025/01/14/1_Week_1.html"><button>Previous Lesson</button></a>    <a href="https://pesalerno.github.io/computing_SP25/main/2025/01/28/3_Week_3.html"><button>Next Lesson</button></a>

<a href="https://pesalerno.github.io/computing_SP25/"><button>Home</button></a>  