---
layout: default
order: 7
title:  "Week 7: Python pt. 2"
date: "2025-02-25"
categories: main
instructor: 
lesson: https://github.com/pesalerno/computing_SP25/blob/main/_posts/2025-02-25-7_Week_7.md
material-type: 
lesson-type: yes
---

<a href="https://pesalerno.github.io/computing_SP25/main/2025/02/18/6_Week_6.html"><button>Previous Lesson</button></a>    <a href="https://pesalerno.github.io/computing_SP25/main/2025/03/04/8_Week_8.html"><button>Next Lesson</button></a>

# Week 7 | Python pt. 2

------------

## Tuesday am: Checking read quality of Illumina data

Let's try to download and install `fastqc`, a program very commonly used as a first step to look into Illumina (or similar next-gen platform) data and try to infer quality of the "run" and library prep. 

First, let's make sure that your linkux machine is up-to-date: 

```bash 
sudo apt update
```

Then, let's try to install fastqc: 

```bash
sudo apt -y install fastqc 
```

To confirm that fastqc has been installed correctly: 

```bash
fastqc --help 
```
And finally, let's run fastqc for all our data files! (For this to work, you either need to be within the folder where your fastq files are located, or add the full/relative path).

```bash
fastqc --noextract -t 2 -o fastqc *fastq.gz
	#did this work correctly for running it on all of your files? where are the output files saved?
```
 
## Understanding more about genomics protocols (library preparations)

We have already discussed a bit about genomics, its inherent difficulties, and a bit also about the Genotyping By Sequencing protocols (as in, the protocol that was used to generate the data we've been working with). So now, we are going to go over a [short lecture](https://docs.google.com/presentation/d/e/2PACX-1vRKqE48P1wS6zrNboqF95FCbUdgr_l8-98W-TTnl-hSJAvICIa3CTrDboBNne0WQoVZY6n1FHmYKaRw/pub?start=false&loop=false&delayms=60000&slide=id.p1) to explain a bit more about these protocols. 

Finally, we will discuss many of the [pitfalls of these kinds of protocols](https://github.com/pesalerno/computing_SP25/blob/main/_files/RADseq-issues.pdf), which more than ever make it clear that bioinformatics needs to be a careful process where the scientific method is applied at every step of the way, and where ***we have to understand the data and the nature of it as much as possible to make sense of the real biological signal behind all the noisy and messy data***. 




## Thursday am: assigned reading discussion

For thursday, [please download and **carefully read the following article**](https://drive.google.com/file/d/1JyL24plknlUw8Qjvpp6MyFeBlsMF6mMB/view?usp=drive_link). This article discusses the importance of Genetic Diversity estimates of non-model organisms and the direct applications to the conservation of at-risk species. 

>------------
> **Before thursday at 8:30am, please read the article and answer the folowing questions**: 
> 
> - What is the main goal of the article? 
> 
> - What is one of the main take-homes of the article? 
> 
> - What is a specific aspect of the article that you found interesting? 
> 
> - If you had lots of money, and time, what you be your chosen species for studying and doing these kinds of approaches, and why? 
> 
> Please submit the answers to these questions using [this google drive folder](https://drive.google.com/drive/folders/1YGl23V35y8azzdeBXHUTb0vr8svV0fgd?usp=sharing), you can submit them by 8:29am on thursday Feb. 27th at the latest :) 
> 
> 
> ------------

## Thrusday pm: more python! 

--------------

<a href="https://pesalerno.github.io/computing_SP25/main/2025/02/18/6_Week_6.html"><button>Previous Lesson</button></a>    <a href="https://pesalerno.github.io/computing_SP25/main/2025/03/04/8_Week_8.html"><button>Next Lesson</button></a>
<a href="https://pesalerno.github.io/computing_SP25/"><button>Home</button></a>  